### DISCUSSION TITLE
Improve the accuracy of the NLP model through user interaction

### DISCUSSION THREAD

**Discussion by KCoderX** [explosion/spaCy#8457] [Help: Best practices]
I am looking for a way to increase the accuracy of my NLP model when my users correct incorrectly extracted information.
The goal is that when the same situation occurs again, the NLP model extracts the information correctly.

I hope I have been able to present my concern in a comprehensible way, and I would therefore be pleased to receive your ideas or tips.

Best regards,
KCoderX

**Comment by polm** [comment]
What you're describing is called "active learning". There's no active learning support built directly into spaCy, but it's one of the core features of [Prodigy](https://prodi.gy/). Basically when you show it data, it'll pick the cases it's least confident about and let you label them.

Letting users label data is certainly another way to do it. You can save user corrections and use them as new training data. You have to be careful if your app is public, as your user's corrections may not be valid. 

Note that even if you do this with spaCy, training works the same as always - after you get new training data you can just retrain your model using the old and new data. Some kinds of machine learning support "online learning", where you can use a single example to update the model on the fly, though this is not usually possible with modern neural models (and isn't supported in spaCy).

**Reply by KCoderX** [reply]
Thank you for the quick response.

That's exactly what I was looking for. Now I have to think about how I can achieve "active learning" indirectly via spaCy.

Do you have any tips on what I should be dealing with when it comes to "active learning"?

**Reply by polm** [reply]
Active learning is a whole research topic, with applications beyond NLP, so there's a lot of material on it. It's not really an introduction to the field, but besides Prodigy, [this demo](https://activelearner.fastforwardlabs.com/) from Fast Forward Labs is cool and maybe a good way to learn some of the basics.
